SingularityUI/app/views/request.coffee

```diff
@@ -61,11 +61,18 @@ class RequestView extends View
             app.router.navigate 'requests', trigger: true
 
     runRequest: (e) =>
-        @model.promptRun =>
-            @trigger 'refreshrequest'
-            setTimeout =>
+        @model.promptRun (data) =>   
+            # If user wants to redirect to a file after the task starts
+            if data.filename.length > 1
+                @listenTo @activeTasks, 'add', @redirectToFile
```

Comment:
is there an alternative that will un-listen once `@redirectToFile` is called? like a `@listenOnce`